--- gnc-html.h	2014-03-01 19:55:35.000000000 +0000
+++ gnc-html.h	2014-05-21 20:08:09.000000000 +0100
@@ -253,19 +253,6 @@
 void gnc_html_set_flyover_cb( GncHtml* html, GncHTMLFlyoverCB newwin_cb, gpointer data );
 void gnc_html_set_button_cb( GncHtml* html, GncHTMLButtonCB button_cb, gpointer data );
 
-/* object handlers deal with <object classid="foo"> objects in HTML.
- * the handlers are looked up at object load time. */
-void gnc_html_register_object_handler( const gchar* classid, GncHTMLObjectCB hand );
-void gnc_html_unregister_object_handler( const gchar* classid );
-
-/* stream handlers load data for particular URLTypes. */
-void gnc_html_register_stream_handler( URLType url_type, GncHTMLStreamCB hand );
-void gnc_html_unregister_stream_handler( URLType url_type );
-
-/* handlers for particular URLTypes. */
-void gnc_html_register_url_handler( URLType url_type, GncHTMLUrlCB hand );
-void gnc_html_unregister_url_handler( URLType url_type );
-
 const gchar* gnc_html_get_embedded_param( gpointer eb, const gchar* param_name );
 
 #endif
