--- methods.defs.orig	2014-05-14 12:15:36.000000000 -0700
+++ methods.defs	2014-05-14 12:15:24.000000000 -0700
@@ -5,6 +5,17 @@
   (parent "GObject")
   (c-name "GncPluginPage")
   (gtype-id "GNC_TYPE_PLUGIN_PAGE")
+  ;; this is not going to work - we have to override the setter functions
+  ;; by writing new functions which directly set the properties
+  ;;(fields
+  ;;  '("gchar*" "page_name")
+  ;;  '("gchar*" "page_long_name")
+  ;;  '("gchar*" "page_color")
+  ;;  '("gchar*" "uri")
+  ;;  '("gchar*" "statusbar_text")
+  ;;  '("gboolean" "use_new_window")
+  ;;  '("gchar*" "ui_description")
+  ;;)
 )
 
 ;; Enumerations and flags ...
