--- gir/QofInstanceType-0.1.gir.orig	2014-09-11 08:50:15.000000000 +0100
+++ gir/QofInstanceType-0.1.gir	2014-09-11 08:51:23.000000000 +0100
@@ -70,11 +70,8 @@
           </parameter>
         </parameters>
       </virtual-method>
-      <property name="book"
-                introspectable="0"
-                writable="1"
-                transfer-ownership="none">
-        <type/>
+      <property name="book" writable="1" transfer-ownership="none">
+        <type name="gpointer" c:type="gpointer"/>
       </property>
       <property name="collection" writable="1" transfer-ownership="none">
         <type name="gpointer" c:type="gpointer"/>
